lib/rubocop/cop/ezcater/graph_ql_type_file_order.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Ezcater
+      class GraphQlTypeFileOrder < Cop
+        ORDERED_DSL_METHODS = %I(
+          extend
+          include
+          description
+          pundit_policy_class
+          pundit_role
+          argument
+          field
+          user_errors_field
+        ).freeze
+
+        LOAD_METHOD_POSITION = ORDERED_DSL_METHODS.length
+        RESOLVE_METHOD_POSITION = ORDERED_DSL_METHODS.length + 1
+
+        METHOD_LOOKUP_LIST = ORDERED_DSL_METHODS.map { |method| ":#{method}" }.join(" | ")
+
+        def on_class(node)
+          gql_offenses = []
+          is_gql_type = false
+
+          type_dsl_calls(node).each_cons(2) do |previous, current|
+            previous = MatchedNode.new(previous)
+            current = MatchedNode.new(current)
+
+            if previous.dsl_method? || current.dsl_method?
+              is_gql_type = true
+            end
+
+            next unless previous > current
+
+            msg = "#{previous.name} should be positioned after #{current.name}"
+            gql_offenses << { node: previous.node, message: msg }
+          end
+
+          if is_gql_type
+            gql_offenses.each do |offense|
+              add_offense(offense[:node], location: :expression, message: offense[:message])
+            end
+          end
+        end
+
+        private # rubocop:disable Lint/UselessAccessModifier
```

Comment:
The rubocop matcher doesn't seem to see the `def_node_search` or the class definition as being "useful". When I added a method here just to test, it stopped complaining.